library/src/main/java/org/apache/apex/malhar/lib/state/managed/ManagedTimeUnifiedStateImpl.java

```diff
@@ -138,6 +139,19 @@ public void purgeTimeBucketsLessThanEqualTo(long timeBucket)
     super.purgeTimeBucketsLessThanEqualTo(timeBucket);
   }
 
+  @Override
+  public void setup(Context.OperatorContext context)
+  {
+    // set UnBoundedTimeBucketAssigner to this managed state impl
+    if (timeBucketAssigner == null) {
+      UnBoundedTimeBucketAssigner unBoundedTimeBucketAssigner = new UnBoundedTimeBucketAssigner();
+      //unBoundedTimeBucketAssigner.setup(context);
```

Comment:
@bhupeshchawda ManagedTimeUnifiedStateImpl is supposed to handle any arbitrary time and the existing time bucket assigner is not efficient for that.